### PR TITLE
Separate out autoconfigure-spi artifact with tracing autoconfiguratio…

### PR DESCRIPTION
**Pull request by anuraaga**
…n SPI.

I have not marked this artifact as alpha since I think it's ready to roll.

It means the same package is in both autoconfigure-spi and autoconfigure - this may cause JMS issues, but I'm not sure it's a big deal while metrics is alpha. If someone knows it is, I'd probably need to temporarily name the metrics SPIs package `autoconfigure.spi.alpha`

### CODE REVIEW

File: docs/apidiffs/current_vs_latest/opentelemetry-extension-trace-propagators.txt [hunk 1]
```diff
@@ -1,2 +1,17 @@
 Comparing source compatibility of  against 
-No changes.
\ No newline at end of file
+***! MODIFIED CLASS: PUBLIC FINAL io.opentelemetry.extension.trace.propagation.B3ConfigurablePropagator  (not serializable)
```
Comment (anuraaga) [inline docs/apidiffs/current_vs_latest/opentelemetry-extension-trace-propagators.txt:2]: Oops - sorry I didn't run a full build on the previous PR

**Comment by jkwatson** [comment]
I'm not an expert, but I thought the Java module system would blow up if two modules tried to export the same package and were being loaded by the same classloader, which these would, I think. :(

File: docs/apidiffs/current_vs_latest/opentelemetry-extension-trace-propagators.txt (hunk 1 above)
Comment (jkwatson) [reply docs/apidiffs/current_vs_latest/opentelemetry-extension-trace-propagators.txt:2]: shame!

**Comment by jkwatson** [comment]
Since we're breaking all of this, we could have `autoconfigure.spi.traces` and `autoconfigure.spi.metrics` to separate the packages. Thoughts?

**Comment by anuraaga** [comment]
Split up, it seems good and does solve the JMS problem.

**Review by jkwatson — APPROVED** [review]
Thanks!

**Review by jsuereth — APPROVED** [review]
In for a penny....

If we're breaking SPI let's make sure we do it right :)

**Pull request merged**